compiler/aarch64/codegen/ARM64BinaryEncoding.cpp

```diff
@@ -359,13 +360,29 @@ uint8_t *TR::ARM64Trg1ImmSymInstruction::generateBinaryEncoding()
    uint8_t *cursor = instructionStart;
    cursor = getOpCode().copyBinaryToBuffer(instructionStart);
    insertTargetRegister(toARM64Cursor(cursor));
-   insertImmediateField(toARM64Cursor(cursor));
 
-   auto label = getLabelSymbol();
-   if (label != NULL)
+   auto sym = getSymbol();
+   if (sym != NULL)
       {
-      cg()->addRelocation(new (cg()->trHeapMemory()) TR::LabelRelative24BitRelocation(cursor, label));
+      if (sym->isLabel())
+         {
+         auto label = sym->getLabelSymbol();
+         if (label != NULL)
+            {
+            cg()->addRelocation(new (cg()->trHeapMemory()) TR::LabelRelative24BitRelocation(cursor, label));
+            }
+         }
+      else if ((getOpCodeValue() == TR::InstOpCode::adr) && sym->isStartPC())
+         {
+         intptr_t offset = reinterpret_cast<intptr_t>(reinterpret_cast<uint8_t *>(sym->getStaticSymbol()->getStaticAddress()) - cursor);
+         if ((offset < -1048576) || (offset >= 1048576))
```

Comment:
Can you add a function like `constantIsSignedImm21()` in ARM64Instruction.hpp, and use it here?

Reply:
Added `constantIsSignedImm21()` to ARM64Instruction.hpp, and changed `TR::ARM64Trg1ImmSymInstruction::generateBinaryEncoding()` to use it.